### ISSUE TITLE
DSL: (user story) explore features from chiventure that can be added to the dsl

### ISSUE THREAD

**Issue by baigh1** [uchicago-cs/chiventure#1182]
team members: alex, william, hira

### **Updated List of Potential Features from Chiventure we Can Add to the DSL**

**- Create a README/User Guide for a Game Developer using a DSL format:**
-- What are the potential elements they can add to the game?
-- A detailed description of the format (tabs, headers, etc).
-- A list of elements (e.g. sound) that the game developer might have wanted to add but are not in development yet.
-- Nested events/complex story structures
-- Basically, updating the (User stories) Wiki Page to be more detailed for ease of User (specific examples related to game development)

**- Integrated NPCs into the DSL**
-- Should create a design document for this to get started

**Comment by wzeng0** [comment]
- Path: 
-- with path direction, destination, conditions, and through. 
--- If paths have different destinations then having a parser would be helpful.
--- parsing paths with multiple conditions would also be helpful with organization

**Comment by wzeng0** [comment]
Alex and I thought of parsing Skill Tree since it included an inventory of skills. An inventory of skills will have a lot of nested structures which would be hard to keep track of. So if we convert it to dsl and wdl then it would be easier to read and keep track of.
- Skill Tree:
- - Inventory
- - - We can parse inventory to be separated to effects and skills. In skills, we have we can parse it's level cap, xp, min_xp, and skill_effect.
- - - - If there are multiple effects for one skill, itll be helpful to parse it.

**Comment by alexlee0422** [comment]
We can possibly implement features relating to quests into DSL. For example, the state as written in quests_state.h and how quests interact with rooms/objects/items/NPCs, etc. Can implement the different parameters of the quests and the reward of completing it etc.

**Comment by wzeng0** [comment]
Alex, Hiri, and I did research on different teams to see which teams can be included in DSL so that we can better maintain the information. We found that we can possible contact the skill-tree, paths, and npc teams to be included.

**Comment by jacknugent1529** [comment]
Issue Score: E

Comments: Nice work on this issue! The updates are informative and the closing statement provides the necessary information. Note that the issue summary should describe your goals with the issue.
